open-sphere-plugins/wfs/src/main/java/io/opensphere/wfs/state/activate/WFSStateActivator.java

```diff
@@ -209,6 +198,17 @@ protected WFSDataTypeBuilder getBuilder()
         return myBuilder;
     }
 
+    /**
+     * Gets all the data groups associated with the list of layer states.
+     *
+     * @param states the layer states
+     * @return the data groups
+     */
+    protected List<DataGroupInfo> getGroupsFromStates(List<? extends WFSLayerState> states)
+    {
+        return states.stream().map(e -> myController.findMemberById(e.getTypeKey()).getParent()).collect(Collectors.toList());

```

Comment:
this makes me a little nervous that findMemberById could return null. what do you think?

Reply:
Addressed- Checking added into call.